### ISSUE TITLE
Including the log message content in the Exception when StreamHandler::write() fails

### ISSUE THREAD

**Issue by ivanhoe011** [Seldaek/monolog#1630]
Currently when `StreamHandler::write()` fails to open the stream it will throw either a `LogicException` or `UnexpectedValueException` with a generic error message, but the log message that we wanted to write to the log will be completely lost in the process. That can be super frustrating, you screw up log permissions without noticing and then days later you see something bad happened but have no idea what, as the log write failed and exception message says just something generic like "Failed to open stream: Permission denied". IMHO dumping the log message as an additional part of the exception would improve the user experience a lot.

So, tl;dr; would it be possible to, on exception in `StreamHandler::write()`, just append `(string) $record['formatted']` to the end of the exception message as an extra piece of info?

**Comment by Seldaek** [comment]
That kinda makes sense yes, but I guess the same logic should be applied to more than just StreamHandler then..
